company/forms.py

```diff
@@ -37,6 +40,33 @@ class Meta:
         ]
 
 
+class MoveProductsForm(SingleButtonMixin, forms.Form):
+    new_company = forms.ModelChoiceField(
+        queryset=Company.objects.all(),
+        widget=Select(),
+        label="Wybierz firmę do której produkty mają być przeniesione",
+    )
+
+    def save(self, pk):
+        failed = []
+        success = []
+        new_company = self.cleaned_data['new_company']
+        if new_company.pk == pk:
```

Comment:
To jest walidacja dwóch zależnych pól: https://docs.djangoproject.com/en/3.2/ref/forms/validation/#cleaning-and-validating-fields-that-depend-on-each-other